keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1767 @@
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Unit tests will be added to verify that the Kubelet's resource and topology
+managers correctly handle the feature gate being enabled or disabled.
+Specifically, tests will cover the conditional logic in the `Admit` and
+`Allocate` functions.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout or rollback can fail in several ways, primarily affecting new or
+restarting pods rather than impacting the resource allocations of already
+running workloads.
+
+*   **Failure Mode: State File Inconsistency** This feature modifies the on-disk
+    state files (e.g., `/var/lib/kubelet/cpu_manager_state`). A Kubelet restart
+    is required for any rollout or rollback, which introduces risks related to
+    state file compatibility.
+
+    *   **During Rollout (Upgrade):** If the new Kubelet version fails to
+        correctly read an old state file, it could crash on startup or
+        misinterpret existing allocations.
+    *   **During Rollback (Downgrade):** An older Kubelet attempting to read a
+        new, pod-aware state file may fail to parse it. This could lead to a
+        Kubelet crash loop or incorrect resource accounting for all `Guaranteed`
+        pods on the node. The established mitigation for such issues is to drain
+        the node, manually delete the state file, and restart the Kubelet.
+
+*   **Impact on Already Running Workloads** Workloads that are already running
+    and stable are **not directly impacted** by the rollout, as their cgroup
+    settings are already configured and will not be changed. The primary risk is
+    associated with the **Kubelet restart** required for the configuration
+    change. The main impact is on pods that are **restarted** during or after
+    the rollout/rollback, as they will be subject to the new (or old) admission
+    logic and could fail to come back up if an issue is present.
+
+###### What specific metrics should inform a rollback?
+
+When the `PodLevelResourceManagers` feature is enabled, operators should closely
+monitor the following metrics. A sustained or significant increase in any of
+these, particularly the new pod-level error metrics, would be a strong indicator
+that a rollback is necessary:
+
+-   **`pod_topology_manager_pinning_errors_total`**: A rise in this metric
+    indicates that the Topology Manager is failing to find suitable NUMA
+    alignments for pods utilizing pod-level resources. This suggests a
+    fundamental issue with the new hint generation or merging logic.
```

Comment:
I think this overlaps with `container_aligned_compute_resources_failure_count` (PTAL to the labels, there's one for scope). Perhaps we can "just" rename this metric, which is currently alpha, and retarget to cover this use case.
It should be fine because the metric is in alpha, let's doublecheck though.
Even if not possible to rename we should probably differentiate this metric more, or reuse the existing one.

Reply:
It would still be relevant to have an specific metric for the pod level resource managers failures, please refer to the proposal made in https://github.com/kubernetes/enhancements/pull/5527#discussion_r2412389247.